core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -944,6 +944,14 @@ public XContentBuilder percentageField(String rawFieldName, String readableField
         return this;
     }
 
+    public XContentBuilder nanosAsMillis(String rawFieldName, String readableFieldName, long nanos) throws IOException {
```

Comment:
The problem is that TImeUnit.toString() is not very accurate, also it rounds to the "larges" unit that needs to be represented, so you get all kinds of variations like "1.1m", "50.9s", "761.5ms", "45585997303" (<- thats nanos then) wheres the current format always prints a fixed-length "ms" with 10 digits precision. I think we need to keep that.